main.ts

```diff
@@ -1,79 +1,104 @@
-import core = require('@actions/core');
-const fetch = require('node-fetch');
-import FormData = require('form-data');
-import fs = require('fs');
-
+import core = require("@actions/core");
+const fetch = require("node-fetch");
+const glob = require("glob");
+import FormData = require("form-data");
+import fs = require("fs");
 
 async function run() {
   try {
     // Get inputs
     const dt_upload_api_key: string = core.getInput("DT_UPLOAD_API_KEY");
     const input_binary_path: string = core.getInput("UPLOAD_BINARY_PATH");
-    core.setSecret(dt_upload_api_key)
-
-    if (!fs.existsSync(input_binary_path)) {
-      throw new Error("Input file does not exist at :" + input_binary_path);
+    // Mask the API key
+    core.setSecret(dt_upload_api_key);
+    // Check that the inputs are set
+    if (!dt_upload_api_key){
+      throw new Error(
+        "DT_UPLOAD_API_KEY must be set!"
+      );
     }
-
-    // retry upload 3 times
-    for (let loop_idx = 0; loop_idx < 3; loop_idx++) {
-
-      // Send the auth request to get the upload URL
-      const auth_response = await fetch(
-        "https://api.securetheorem.com/uploadapi/v1/upload_init",
-        {
-          method: 'POST',
-          headers: {
-            Authorization: "APIKey " + dt_upload_api_key,
-            Accept: "application/json",
-            "Content-Type": "application/json",
-          },
-        }
+    if (!input_binary_path){
+      throw new Error(
+        "UPLOAD_BINARY_PATH must be set!"
       );
+    }
 
-      let auth_json
-      try {
-        auth_json = await auth_response.json()
-      } catch (err) {core.setFailed(err);}
+    const files = glob.sync(input_binary_path);
+    if (!files.length) {
+      throw new Error(
+        "Did not find any files that match path:" + input_binary_path
+      );
+    }
+    console.log("Found files matching path: " + input_binary_path);
+    console.log(files);
 
-      if (auth_response.status !== 200) {
-        // handles auth failure
-        core.setFailed(auth_json);
-        break;
+    // Upload all the files that matched the file path
+    let output: Array<any> = []
+    for (const file_path of files) {
+      if (!fs.existsSync(file_path)) {
+        throw new Error("Could not find file:" + file_path);
       }
+      // retry upload 3 times
```

Comment:
There are two instances of this code in the PR (the "retry upload 3 times" logic); we should only have one, I think?

Reply:
you were confused i think by the fact that there is a ts and js file. However it is only present once in the TS file (and a second time on the generated js code)

Reply:
you were confused i think by the fact that there is a ts and js file. However it is only present once in the TS file (and a second time on the generated js code)